lib/src/builder.dart

```diff
@@ -135,6 +136,20 @@ class AggregateTestBuilder extends Builder {
     // Don't generate an empty aggregate test.
     if (imports.isEmpty) return;
 
+    if (_config.randomizeAggregation) {
+      log.fine('randomizing test order');
+      final randomizedMains = mains.toString().split(';\n')
+        ..removeWhere((element) => element.isEmpty);
```

Comment:
Instead of writing the mains to a StringBuffer above (which then forces us to split them back up into a list, clear the buffer, and rewrite it in order to handle the randomization), it might be simpler to make `mains` above a List that we add the `  $prefix.main();` strings to. That way, we can randomize the list here when enabled and then below we can update the lines that write the full file contents to something like this:

```dart
    final contents = DartFormatter().format('''@TestOn('browser')
import 'package:test/test.dart';
$imports
void main() {
${mains.join('\n');}
}
''');
```